add docs png image optimizer workflow

escherize opened the pull request:
This runs when there is a png file in /docs commited. For all such files, when it can be optimized via ImageMagick's `mogrify`, adds it to the PR as a commit from "Metabase bot".

romeovs committed:
Preview native query with snippet (#60559)

* Allow previewing query that only has snippets

* Add e2e test

alxnddr committed:
show active y-axis split lines on hover (#60238)

* show y-axis split line on hover

* fix axis split line visibility on hover

* review

jeff-bruemmer committed:
docs - compress images (#60579)

* compress images

* more images

escherize committed:
add image optimizer

- This runs when there is a png file in /docs commited. For all such
files, when it can be optimized via ImageMagick's `mogrify`, adds it to
the PR.

escherize committed:
tweak script

jeff-bruemmer commented on .github/workflows/optimize-images.yml:
```diff
@@ -0,0 +1,76 @@
+        echo "Optimizing PNG files..."
+        for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+          echo "Processing: $file"
+          mogrify -quality 100 "$file"
```
```suggestion
          mogrify -strip -quality 95 "$file"
```

alxnddr committed:
filters in dashcards tracking (#60517)

* filters in dashcards tracking

* update tracking according to spec

escherize committed:
Update .github/workflows/optimize-images.yml

Co-authored-by: Jeff Bruemmer <jeff.bruemmer@gmail.com>

escherize committed:
Merge remote-tracking branch 'origin/master' into optimize-docs-images

escherize committed:
add test png

improve workflow output

properly detect staged changes instead of working directory changes

remove test png

readd test png

fix glob expansion

remove test png

readd test png

fix script

bump

switch to push

update change detection

rename unmogrified test.png -> test2.png

git add the staged files directly (shell expansion?)

docs/test2.png -> docs/test3.png

match the username + email to CLA-signed 'user'

add test5 (unmogged)

Auto-optimize PNG images with ImageMagick

escherize committed:
add an uncompressed image

escherize committed:
finally fix optimize images script

escherize committed:
add yet another image

github-automation-metabase committed:
Auto-optimize PNG images with ImageMagick

escherize commented:
workflow passing here: https://github.com/metabase/metabase/actions/runs/16181177797/job/45677957050  -- creates this commit: [c9914dc](https://github.com/metabase/metabase/pull/60581/commits/c9914dc9c5f6c2b138e4fc56eec858126874066a)

escherize committed:
remove test image

escherize committed:
Merge branch 'master' into optimize-docs-images

github-automation-metabase committed:
Auto-optimize PNG images with ImageMagick

escherize committed:
switch to pngcrush

escherize committed:
test nine

escherize committed:
Merge branch 'optimize-docs-images' of github.com:metabase/metabase into optimize-docs-images

github-automation-metabase committed:
Auto-optimize PNG images with pngcrush

escherize committed:
update workflow + churn images

escherize committed:
Merge branch 'optimize-docs-images' of github.com:metabase/metabase into optimize-docs-images

github-automation-metabase committed:
Optimized 2 images, saved 260.3 KB

escherize committed:
include % difference when possible

escherize committed:
Merge branch 'optimize-docs-images' of github.com:metabase/metabase into optimize-docs-images

escherize committed:
bump images

github-automation-metabase committed:
Optimized 2 images, saved 333.0 KB (46.1% smaller)

escherize committed:
remove test pics

escherize committed:
Merge branch 'optimize-docs-images' of github.com:metabase/metabase into optimize-docs-images
